common/pinot/pinotQueryValidator.go

```diff
@@ -326,3 +346,26 @@ func processCustomString(comparisonExpr *sqlparser.ComparisonExpr, colNameStr st
 	return fmt.Sprintf("(JSON_MATCH(Attr, '\"$.%s\" is not null') "+
 		"AND REGEXP_LIKE(JSON_EXTRACT_SCALAR(Attr, '$.%s', 'string'), '%s*'))", colNameStr, colNameStr, colValStr)
 }
+
+func trimTimefieldValueFromNanoToMilliSeconds(original *sqlparser.SQLVal) (*sqlparser.SQLVal, error) {
+	// Convert the SQLVal to a string
+	valStr := string(original.Val)
+	// Convert to an integer
+	valInt, err := strconv.ParseInt(valStr, 10, 64)
+	if err != nil {
+		return original, fmt.Errorf("error: failed to parse int from SQLVal")
```

Comment:
include `valStr` in the error otherwise error message is not useful to troubleshoot